tools/createSovereignGenesis/create-sovereign-genesis.ts

```diff
@@ -145,6 +167,52 @@ async function main() {
         outWETHAddress = wethObject.address;
     }
 
+    // check preMintAddress is an address
+    if(ethers.isAddress(createGenesisSovereignParams.preMintAddress) == false){
+        logger.error('preMintAddress: not a valid address');
+        throw new Error('preMintAddress: not a valid address');
+    }
+
+    // add preMint address & preMint value
+    logger.info('Add preMintAddress and preMintAmount');
+    finalGenesis.genesis.push({
+        accountName: "preMint Account",
+        balance: BigInt(createGenesisSovereignParams.preMintAmount).toString(),
+        address: createGenesisSovereignParams.preMintAddress,
+    });
+
+    // set timelock storage
+    logger.info('Add timelock setup');
+    const timelockContractInfo = finalGenesis.genesis.find(function (obj) {
+        return obj.contractName === "PolygonZkEVMTimelock";
+    });
+
+    const storageTimelock = initializeTimelockStorage(
+        createGenesisSovereignParams.minDelayTimelock,
+        createGenesisSovereignParams.timelockAdminAddress,
+        timelockContractInfo.address
+    );
+
+    timelockContractInfo.storage = storageTimelock;
+
+    // regenerate root with the zkEVM root
+    const poseidon = await getPoseidon();
+    const {F} = poseidon;
+
+    const zkEVMDB = await ZkEVMDB.newZkEVM(
+        new MemDB(F),
+        poseidon,
+        [F.zero, F.zero, F.zero, F.zero],
+        [F.zero, F.zero, F.zero, F.zero],
+        finalGenesis.genesis,
+        null,
+        null,
+        createGenesisSovereignParams.chainID
+    );
+
+    // update genesis root
+    finalGenesis.root = smtUtils.h4toString(zkEVMDB.getCurrentStateRoot());
```

Comment:
i would update this to, zkevmRoot, since this root won't be have much porpuses from now ( even we can delete it D:

Reply:
I asked Jerry if this check is still valid on the `cdk-sovereign-erigon`. If that is the case, I would not remove it.
Not sure if any external uses this genesis file, so a renamed could potentially break it.

I would wait to have a confirmation if it is used --> then I would remove it.
Otherwise, I will keep it until we can remove it with no renaming